ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -1219,11 +1223,21 @@ private void handleMetadataBlob(BlobData blobData, byte[] userMetadata, ByteBuff
       chunkMetadataList = compositeBlobInfo.getChunkMetadataList();
       boolean rangeResolutionFailure = false;
       try {
+        if (options.getBlobOptions.hasBlobSegment()) {
+          int requestedSegment = options.getBlobOptions.getBlobSegmentIdx();
+          if (requestedSegment < 0 || requestedSegment >= chunkMetadataList.size()) {
+            throw new IllegalArgumentException(
+                "Bad segment number: " + requestedSegment + ", num of keys: " + chunkMetadataList.size());
+          }
+          chunkMetadataList = chunkMetadataList.subList(requestedSegment, requestedSegment + 1);
```

Comment:
so here we use `subList` rather than normal list.get() is to avoid extra copy?

Reply:
the intent is to populate or assign `chunkMetadataList` which is a list object so list.get() wouldn't work here